datafusion/src/execution/context.rs

```diff
@@ -125,12 +127,26 @@ pub struct ExecutionContext {
     pub state: Arc<Mutex<ExecutionContextState>>,
 }
 
+lazy_static! {
```

Comment:
I don't understand the need for a global static execution context. I think the intent is that the user of DataFusion would determine if it wanted to use a single context or if it wanted to re-use the same context

Reply:
I made a registry (a hashmap) for storing all available protocol handlers (or ObjectStore as you suggested above) in the context at runtime and registering the LocalFS handler by default. In this way, I intended to make DataFusion’s core protocol agnostic and get different handlers according to the scheme of a path when users are creating new tables at runtime (based on their previous registrations), and get the suitable handler for each table.

And at the same time, I thought a global/sharable context is useful when we consider controlling the total amount of memory Datafusion could use. All physical operators should register their memory usage with a global context, and try to allocate more memory or spill itself if used up all allowed memory.

Reply:
I agree a global context may be interesting to to add. It doesn't seem entirely related to reading external data sources, however -- I suggest proposing that change in a separate PR to make sure discussion on it isn't lost in the discussion on this PR

Reply:
I've filed #824 for this and will remove it from this PR.

Reply:
I agree with @alamb on this one, would be better to have discussion about global static in a separate PR. This way, it will be easier for us to approve and merge code change specific to remote data source.

Reply:
thank you @yjshen